### ISSUE TITLE
Fresh 1.0.3 install skips userdata copy/volume expansion step

### ISSUE THREAD

**Issue by mennoliefstingh** [rg35xx-cfw/Koriki#96]
### What happened?

I've been troubleshooting a fresh install using `batocera-rg35xx-rg35xx--20240103.img` (1.0.3) for a few hours now, until I read that there should be a 'copying userdata' step. This step did not take place when flashed a fresh SD card with Balena on macOS. 

I just flashed the 1.0.0 release and now saw the copying/expansion step for the first time; this is also a valid workaround but something that should probably be addressed in a future release.

**Comment by mennoliefstingh** [comment]
Didn't remember to look for logs (sorry) but I do have a diskutil terminal output that might be relevant: 

```diskutil list
/dev/disk0 (internal, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      GUID_partition_scheme                        *500.3 GB   disk0
   1:             Apple_APFS_ISC Container disk1         524.3 MB   disk0s1
   2:                 Apple_APFS Container disk3         494.4 GB   disk0s2
   3:        Apple_APFS_Recovery Container disk2         5.4 GB     disk0s3

/dev/disk3 (synthesized):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      APFS Container Scheme -                      +494.4 GB   disk3
                                 Physical Store disk0s2
   1:                APFS Volume Macintosh HD            9.8 GB     disk3s1
   2:              APFS Snapshot com.apple.os.update-... 9.8 GB     disk3s1s1
   3:                APFS Volume Preboot                 6.1 GB     disk3s2
   4:                APFS Volume Recovery                1.7 GB     disk3s3
   5:                APFS Volume Data                    452.2 GB   disk3s5
   6:                APFS Volume VM                      5.4 GB     disk3s6

/dev/disk4 (internal, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:     FDisk_partition_scheme                        *63.9 GB    disk4
   1:             Windows_FAT_32 BATOCERA                3.2 GB     disk4s1
   2:                 Linux_Swap                         536.9 MB   disk4s2
   3:                      Linux                         536.9 MB   disk4s3
                    (free space)                         59.6 GB    -
                   ```

**Comment by acmeplus** [comment]
The step takes place but it's not displayed on the screen. However that version has an issue with the SHARE partition being formatted in ext4 format. The new v1.0.3.1 update fixes that.

**Comment by Nexus6p** [comment]
It expands on 1.03.1 as far as I can tell, but SHARE partition is still ext4

**Comment by Al3xJ0bs** [comment]
1.0.3.1 still does not expand the SHARE partition ( Takes up 512 mb) and is still in EXT4 format

**Comment by gordoneyal** [comment]
For me too, after the 1.0.3.1 image is flashed, a 512MB EXT4 format SHARE partition apears.
However, when I reformat this partition to FAT32, it expands **automatically** to 26GB, during first boot.

**Comment by Al3xJ0bs** [comment]
> For me too, after the 1.0.3.1 image is flashed, a 512MB EXT4 format SHARE partition apears. However, when I reformat this partition to FAT32, it expands **automatically** to 26GB, during first boot.

It works

**Comment by manumito** [comment]
> For me too, after the 1.0.3.1 image is flashed, a 512MB EXT4 format SHARE partition apears. However, when I reformat this partition to FAT32, it expands **automatically** to 26GB, during first boot.

It worked for me too.

**Comment by Dylan96** [comment]
I just used version 1.0.3.1 and my SHARE partition is still formatted as EXT4
![partitions](https://github.com/rg35xx-cfw/Koriki/assets/1145144/0c15aa32-3fbd-49d4-8074-734726ffaccc)

**Comment by souporman** [comment]
> I just used version 1.0.3.1 and my SHARE partition is still formatted as EXT4 ![partitions](https://private-user-images.githubusercontent.com/1145144/303908567-0c15aa32-3fbd-49d4-8074-734726ffaccc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1NjY0NzEsIm5iZiI6MTcwOTU2NjE3MSwicGF0aCI6Ii8xMTQ1MTQ0LzMwMzkwODU2Ny0wYzE1YWEzMi0zZmJkLTQ5ZDQtODA3NC03MzQ3MjZmZmFjY2MucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDRUMTUyOTMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDFjZjExMzA2OWI3YmU4NGI4OGQ4OWI0NDgyMGY0MmZhZjA0ZDBmMTk3N2I3YTUzNjU4MjRhZmM4NTlhZDg5MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.qZYINkRGNVmPlK5sFlWcDYwBpDwI0Xp3NR29GcO62nk)

Yeah, this is a bug, but you can workaround it by manually reformatting that SHARE partition from EXT4 to FAT32 directly after you flash it.  After that, pop it in your handheld and boot it up once.  It'll expand SHARE and copy over the correct data and you'll be able to see it on Windows to add your games/Bios.
